### PR TITLE
DB

### PR DESCRIPTION
**Pull request by 1lyan**

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -35,31 +43,40 @@ const listenMessage = `Listening on port: ${config.APP_PORT}. ENV is ${config.EN
...
+    collection.insert(movie);
     return response.status(200).json({
       message: `New movie added: ${name}`,
       data: movie,
     });
   }
 
   // movie already exists in DB
-  if (movieStore.exist(data["imdbID"])) {
+  if (collection.findOne({ imdbID: data["imdbID"] })) {
```
Comment (rfist) [inline src/server.ts]: You can't use async operation `findOne` in if condition. You should make search first and save it result somewhere
```
const movie = await collection.findOne({ imdbID: data["imdbID"] });

if (movie) {....}
```

File: src/server.ts [hunk 2]
```diff
@@ -35,31 +43,40 @@ const listenMessage = `Listening on port: ${config.APP_PORT}. ENV is ${config.EN
...
       message: `Movie already exist in DB: ${name}`,
     });
   }
 
   // movie is found in IMDB but not in local DB - so let's add it
-  if (!movieStore.exist(data["imdbID"])) {
+  if (!collection.findOne({ imdbID: data["imdbID"] })) {
     const movie = { ...data, comment, personalScore };
-    movieStore.add(movie);
+    collection.insert(movie);
```
Comment (rfist) [inline src/server.ts]: This one need `await` operator. And you can see warning in your logs:
[MONGODB DRIVER] Warning: collection.insert is deprecated. Use insertOne, insertMany or bulkWrite instead.
(Use `node --trace-warnings ...` to show where the warning was created)

File: src/server.ts [hunk 3]
```diff
@@ -72,57 +89,77 @@ app.post("/movies", async (request: Request, response: Response) => {
...
   if (personalScore) {
-    movie["personalScore"] = personalScore;
+    updateFields["personalScore"] = personalScore;
   }
 
+  await collection.updateOne({ imdbID: id }, { $set: updateFields });
+  console.log("MOVIE", movie);
   return response.status(200).json({
     message: `Movie updated: ${movie.Title}`,
     data: movie,
```
Comment (rfist) [inline src/server.ts]: You are returning old version of movie, without updated fields.

File: src/server.ts [hunk 4]
```diff
@@ -72,57 +89,77 @@ app.post("/movies", async (request: Request, response: Response) => {
...
   });
 });
 
-app.delete("/movies/:id", (request: Request, response: Response) => {
+app.delete("/movies/:id", async (request: Request, response: Response) => {
+  const client = await MongoClient.connect(DB_URL);
```
Comment (rfist) [inline src/server.ts]: You create new mongodb client for every single endpoint and for every request, it's not good.
 ```
  const client = await MongoClient.connect(DB_URL);
  const db = client.db("moviesDB");
  const collection = db.collection("movies");
  ```
Move this code to line 16, and then just call `collection` when you need it.

File: src/server.ts [hunk 5]
```diff
@@ -72,57 +89,77 @@ app.post("/movies", async (request: Request, response: Response) => {
...
+  const movie = await collection.findOne({ imdbID: id });
 
   if (!movie) {
     return response.status(404).json({
       message: "Movie not found",
     });
   }
 
-  movieStore.destroy(movie);
+  collection.deleteOne({ imdbID: id });
```
Comment (rfist) [inline src/server.ts]: Missed `await`

File: src/server.ts [hunk 6]
```diff
@@ -2,10 +2,18 @@ import express from "express";
...
+  noAppPortMsg,
+  noEnvMsg,
+  noApiKeyMsg,
+  defaultScore,
+  DB_URL,
+} from "./constants";
 import OmdbService from "./services/omdbService";
 import MovieStore from "./stores/movieStore";
 
+const MongoClient = require("mongodb").MongoClient;
```
Comment (rfist) [inline src/server.ts]: You can get benefits from typescript, if you do import properly.
```
import * as mongoDB from "mongodb";
const MongoClient = mongoDB.MongoClient;
```
Now you have autocomplete and correct return types.

File: src/server.ts [hunk 7]
```diff
@@ -35,31 +43,40 @@ const listenMessage = `Listening on port: ${config.APP_PORT}. ENV is ${config.EN
...
     // movie not found in IMDB - let's add it to local DB
-    const movie = { imdbID: movieStore.nextId(), Title: name, comment, personalScore };
-    movieStore.add(movie);
+    const movie = {
+      imdbID: movieStore.nextId(),
+      Title: name,
+      comment,
+      personalScore,
+    };
+    collection.insert(movie);
```
Comment (rfist) [inline src/server.ts]: Missed `await`

**Review by rfist — CHANGES_REQUESTED** [review]

File: src/server.ts (hunk 4 above)
Comment (1lyan) [reply src/server.ts]: @rfist, won't work because of 
```
const client = await MongoClient.connect(DB_URL);
```

File: src/services/db.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { DB_URL } from "../constants";
+
+const MongoClient = require("mongodb").MongoClient;
```
Comment (rfist) [inline src/services/db.ts:3]: Still can't understand, why are using `require` in typescript project :disappointed:

**Review by rfist — APPROVED** [review]
Good job!

**Pull request merged**